### PR TITLE
Update dependencies and fix cs

### PR DESCRIPTION
**Pull request by mangoischke**
| Q             | A
| ------------- | --------------------------------------------------------------------- |
| Bug fix?      | no                                                                |
| New feature?  | no |
| Improvement?  | no   |
| BC breaks?    | no                                                                |
| Deprecations? | no    |
| Docs PR       | **missing**    |

### CODE REVIEW

File: src/Model/ModelInterface.php [hunk 1]
```diff
@@ -10,37 +10,31 @@ interface ModelInterface
...
+    public function remove (object $entity) : self;
 
 
     /**
      * Flushes the entity changes to the database.
      *
      * @return $this
      */
-    public function flush ();
+    public function flush () : self;
```
Comment (apfelbox) [inline src/Model/ModelInterface.php]: This change is wrong, unfortunately.

2019-11-25 at 12 57

We are not returning the static `self`, but rather the dynamic `static` (look up "late static binding" in PHP as reference / explanation). So in my `TestModel` your code says that we are returning a (static determined) `ModelInterface`, although the code actually returns a `TestModel`. You can see that the `hello` method is therefore missing.

The 100% correct code would be sth like `public function flush () : static;` or even `public function flush () : this;`, however both of these are not (yet?) supported in PHP.

Although this change is technically valid, it is semantically invalid, as we lose a lot of information. And having proper auto completion here is more important than some type checks.

tl;dr please remove.

**Comment by apfelbox** [comment]
Superseded by #64
